### DISCUSSION TITLE
Advice: once macOS apps are installed, do you let them update on their own or via Homebrew only?

### DISCUSSION THREAD

**Discussion by YomsGH** [Homebrew/discussions#3548] [Getting started]
Hi,

I'm going to reinstall my Mac from scratch and use Homebrew to install my apps (1password, Signal desktop, Proton, etc.)

The main reason is that Homebrew also allows me to make clean delete of apps that I no longer want and not rely on softwares such as "AppCleaner". Well, at least that's what I hope to achieve.

My question is: once an app is installed, should I therefore disable auto-update within the app? Or can I let the app update on its own? If I let it update on its own, will Homebrew be able to cleanly remove an app that it has installed, but that has been updated on its own? Or will Homebrew somehow get lost?

Thanks.

**Comment by p-linnane** [comment]
You can handle this however you prefer. Casks that indicate they auto update will not show up in your list of outdated packages unless you run `brew outdated --greedy`. When you want to remove an app just run `brew uninstall --cask --zap foo` to dispatch the `zap` stanza which will try to remove all preferences and other associated files.

My personal take is that I keep everything up to date using `brew`, and do not use app autoupdates. It's a personal preference thing.

**Reply by YomsGH** [reply]
Thanks for your feedback. If the app auto-updates on its own, by what magic is Homebrew able to remove all files when uninstalling an app, especially if updates have installed newer files in new directories?

Basically, what I want above all is a solution that allows to properly uninstall apps. I thought using Homebrew rather than downloading installer from websites, then use a cleaner was a better idea.

**Reply by p-linnane** [reply]
When you update Homebrew you will still receive the latest version of the cask definition Ruby riles (`foo.rb`). That should account for newly changed files as users update the casks on Homebrew.

Alternatively, you can use something like CleanMyMac X to try to remove all extra files when you are uninstalling.

**Reply by YomsGH** [reply]
Just to be sure to understand you perfectly, I still receive the latest version of the cask definition even if I previously updated the app through its internal auto update?

**Reply by p-linnane** [reply]
Yes. When you run `brew update` it pulls down the latest from GitHub regardless of whether you actually update the applications or not.

**Reply by YomsGH** [reply]
Thanks, it's pretty clear now
